MaintenanceManager/MaintenanceManager.cpp

```diff
@@ -813,6 +800,8 @@ namespace WPEFramework {
                     string abort_flag="";
 
                     /* only one maintenance at a time */
+                    /* Lock so that g_notify_status will not be updated  further */
+                    m_statusMutex.lock();
```

Comment:
Why are we locking at 2 places? here and within onMaintenanceStatusChange() before updating g_notify_status. Once this lock is taken here, it's going to block the task_execution_thread from updating g_notify_status.

Reply:
Seems fine to me as the start method will return with in progress error. If start method was able to acquire lock then the previous execution should have been completed. Let me know if you still see some race condition here, we can discuss.